diff --git a/dwm.c b/dwm.c
index 169adcb..5067cc3 100644
--- a/dwm.c
+++ b/dwm.c
@@ -980,11 +980,19 @@ keypress(XEvent *e) {
 
 	ev = &e->xkey;
 	keysym = XKeycodeToKeysym(dpy, (KeyCode)ev->keycode, 0);
-	for(i = 0; i < LENGTH(keys); i++)
+	for(i = 0; i < LENGTH(keys); i++) {
 		if(keysym == keys[i].keysym
+        && keys[i].mod
 		&& CLEANMASK(keys[i].mod) == CLEANMASK(ev->state)
-		&& keys[i].func)
+		&& keys[i].func) {
 			keys[i].func(&(keys[i].arg));
+            return;
+        }
+        if(!keys[i].mod && keysym == keys[i].keysym) {
+			keys[i].func(&(keys[i].arg));
+            return;
+        }
+    }
 }
 
 void
