--- a/dwm.c	2025-06-26 13:33:50.810284569 -0700
+++ b/dwm.c	2025-06-26 13:34:58.473676639 -0700
@@ -1002,6 +1002,21 @@
 #endif /* XINERAMA */
 
 void
+keysh(const Arg *arg)
+{
+    if (fork() == 0){
+		if (dpy)
+			close(ConnectionNumber(dpy));
+		setsid();
+        char *cmd[] = {keyshpath, arg->v, NULL};
+		execvp(keyshpath, (char **)cmd);
+		fprintf(stderr, "dwm: execvp %s", ((char **)arg->v)[0]);
+		perror(" failed");
+		exit(EXIT_SUCCESS);
+    }
+}
+
+void
 keypress(XEvent *e)
 {
 	unsigned int i;
@@ -1014,7 +1029,15 @@
 		if (keysym == keys[i].keysym
 		&& CLEANMASK(keys[i].mod) == CLEANMASK(ev->state)
 		&& keys[i].func)
-			keys[i].func(&(keys[i].arg));
+            if (keys[i].func == keysh) {
+                char *strkey = malloc(10);
+                sprintf(strkey, "0x%x", keysym);
+                Arg arg = {.v = strkey};
+                keysh(&(arg));
+                free(strkey);
+            }
+            else
+                keys[i].func(&(keys[i].arg));
 }
 
 void
